src/attributes/classes.js

```diff
@@ -26,10 +30,9 @@ jQuery.fn.extend({
 
 			for ( ; i < len; i++ ) {
 				elem = this[ i ];
-				cur = elem.nodeType === 1 && ( elem.className ?
-					( " " + elem.className + " " ).replace( rclass, " " ) :
-					" "
-				);
+				curValue = getClass( elem );
+				cur = elem.nodeType === 1 &&
```

Comment:
Actually, just tried this change and got a failure because cur retains its value from the previous loop and needs to be set to something falsey.